src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/DisablePrivateReflectionAttribute.cs

```diff
@@ -3,6 +3,7 @@
 
 namespace System.Runtime.CompilerServices
 {
+    [Obsolete("DisablePrivateReflectionAttribute has no effect in .NET Core and .NET 5.0+ applications.")]
```

Comment:
Per https://github.com/dotnet/runtime/issues/43247#issuecomment-713070795, this needs an ID set on it. That ID will need to be recorded in https://github.com/dotnet/runtime/blob/0c3cc15f90f0e3b81ce3199be1ac217564043635/src/libraries/Common/src/System/Obsoletions.cs.  You can see other examples in that file and how they're used in call sites.

In addition, this is editing the src file, but the class in the reference assembly file https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Runtime/ref/System.Runtime.cs will also need to have the attribute applied.